### PR TITLE
Add cdk-addons=true label to all templates and select on that

### PR DESCRIPTION
**Pull request by Cynerva**
This is a follow-up to #20, which doesn't quite work right when `enable-dashboard-addons=false`, for two reasons:

1. With no `--prune=true` when applying the `k8s-app=kubernetes-dashboard` group, the dashboard addons are never removed.
2. Even with `--prune=true`, `kubectl apply` blows up when no addons are selected:

```
2017-10-18 19:25:34 DEBUG update-status error: no objects passed to apply
2017-10-18 19:25:34 DEBUG update-status Traceback (most recent call last):
2017-10-18 19:25:34 DEBUG update-status   File "/snap/cdk-addons/x1/apply", line 102, in <module>
2017-10-18 19:25:34 DEBUG update-status     main()
2017-10-18 19:25:34 DEBUG update-status   File "/snap/cdk-addons/x1/apply", line 13, in main
2017-10-18 19:25:34 DEBUG update-status     apply_addons()
2017-10-18 19:25:34 DEBUG update-status   File "/snap/cdk-addons/x1/apply", line 73, in apply_addons
2017-10-18 19:25:34 DEBUG update-status     kubectl(*args)
2017-10-18 19:25:34 DEBUG update-status   File "/snap/cdk-addons/x1/apply", line 82, in kubectl
2017-10-18 19:25:34 DEBUG update-status     return subprocess.check_output(cmd)
2017-10-18 19:25:34 DEBUG update-status   File "/usr/lib/python3.5/subprocess.py", line 626, in check_output
2017-10-18 19:25:34 DEBUG update-status     **kwargs).stdout
2017-10-18 19:25:34 DEBUG update-status   File "/usr/lib/python3.5/subprocess.py", line 708, in run
2017-10-18 19:25:34 DEBUG update-status     output=stdout, stderr=stderr)
2017-10-18 19:25:34 DEBUG update-status subprocess.CalledProcessError: Command '['/snap/cdk-addons/x1/kubectl', 'apply', '-f', '/root/snap/cdk-addons/x1/addons', '--recursive', '--namespace=kube-system', '-l', 'k8s-app=kubernetes-dashboard', '--prune=true', '--force']' returned non-zero exit status 1
```

The proposed fix in this PR is to remove our dependence on the `k8s-app=kubernetes-dashboard` and `kubernetes.io/cluster-service=true` labels, and instead apply our own `cdk-addons=true` label to all addon definitions. Then we can use that instead.

@ktsakalozos can you take a look at this? Thanks.

### CODE REVIEW

File: cdk-addons/apply [hunk 1]
```diff
@@ -43,33 +44,34 @@ def render_template(file, context, required=True):
         return
     with open(source) as f:
         template = Template(f.read())
+    content = template.render(context)
+
+    # apply cdk-addons=true label
+    data = [yaml.load(part) for part in content.split('\n---') if part.strip()]
```
Comment (marcoceppi) [inline cdk-addons/apply]: `data = yaml.load_all(content)` might be faster

File: cdk-addons/apply [hunk 2]
```diff
@@ -43,33 +44,34 @@ def render_template(file, context, required=True):
...
+    for part in data:
+        part["metadata"]["labels"]["cdk-addons"] = "true"
+    content = '\n---\n'.join(yaml.dump(part) for part in data)
```
Comment (marcoceppi) [inline cdk-addons/apply]: Likewise, `yaml.dump_all` might be more appropriate

**Comment by ktsakalozos** [comment]
Thank you for this PR. This is something I did not test.

Since we are changing the label acting in the prune selector how would we update any already existing deployments? Based on what is here https://kubernetes.io/docs/user-guide/kubectl/v1.8/#apply we would need to include the old label to delete any matching services, right?

**Comment by Cynerva** [comment]
Updated to use yaml.load_all and yaml.dump_all, thanks @marcoceppi :+1:

@ktsakalozos You're right, there is a problem with the upgrade scenario. Most addons are fine; since the resource names don't change, `kubectl apply` updates them to have the `cdk-addons=true` label and everything is fine from there.

However, going from 1.8.0 to 1.8.1, the heapster addon changes name:

```
$ kubectl get deployment --all-namespaces
NAMESPACE     NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
kube-system   heapster-v1.4.3          1         1         1            1           13m
kube-system   heapster-v1.5.0-beta.0   1         1         1            1           41m
kube-system   kube-dns                 1         1         1            1           41m
kube-system   kubernetes-dashboard     1         1         1            1           41m
```

Argh! Off to think of a solution.

**Comment by Cynerva** [comment]
Updated to add the `kubernetes.io/cluster-service=true` label and continue to use that in our call to `kubectl apply`, for now. That should handle pruning properly in the upgrade case.

After the `cdk-addons=true` label has been live for a couple releases, we can switch to selecting on that and stop adding/using `kubernetes.io/cluster-service=true`.

**Comment by ktsakalozos** [comment]
LGTM +1

**Pull request merged**